Use newer method overloads in auth handlers

martincostello opened the pull request:
**PR Title**

Use newly added overloads of already-used methods in authentication handlers.

**PR Description**

I noticed that some of the new overloads added in .NET 5.0 weren't used, so I've switched various usages within the Authentication part of the solution to use them. I also applies some minor improvements in the files I touched that Visual Studio suggested.

* Use new overloads of methods added in .NET 5.0 that accept a `CancellationToken`.
* Use `WriteAsync()` overload that accepts a span ~and a `CancellationToken`~, instead of an array and indexes.
* Use `StringBuilder.Append(char)` instead of `StringBuilder.Append(string)` when there is only a single character.
* Make some methods that don't access instance data `static`.
* Remove some unused code.
* Use compound assignment.
* Fix two incorrect test names from copy-paste.

martincostello committed:
Use new CancellationToken overloads

Use new overloads of methods added in .NET 5.0 that accept a CancellationToken.

martincostello committed:
Use StringBuilder.Append(char)

Use StringBuilder.Append(char) instead of StringBuilder.Append(string) when there is only a single character.

martincostello committed:
Use WriteAsync(Span<byte>, CancellationToken)

Use WriteAsync() overload that accepts a span and a CancellationToken, instead of an array and indexes to write all bytes.

martincostello committed:
Make methods static

Make methods that do not access instance data static.

martincostello committed:
Remove unused code

Remove an unused method and an unused parameter.

martincostello committed:
Use compound assignment

Use the compound assignment operator.

martincostello committed:
Fix test names

Fix copy-paste from Facebook tests.

Tratcher commented on src/Security/Authentication/OpenIdConnect/src/OpenIdConnectHandler.cs:
```diff
@@ -276,7 +276,7 @@ public async virtual Task SignOutAsync(AuthenticationProperties properties)
                 Response.Headers[HeaderNames.Expires] = HeaderValueEpocDate;
 
-                await Response.Body.WriteAsync(buffer, 0, buffer.Length);
+                await Response.Body.WriteAsync(buffer, Context.RequestAborted);
```
It's better not to pass the CT to WriteAsync. The write will fail silently if the client disconnects, but if you pass the CT then it will throw. We don't care if this write succeeds or fails.

martincostello committed:
Remove CancellationToken

Remove the CancellationToken from the WriteAsync() call to address review feedback.

Tratcher commented:
Thanks

davidfowl commented on src/Security/Authentication/OpenIdConnect/src/OpenIdConnectHandler.cs:
```diff
@@ -105,7 +105,7 @@ protected virtual async Task<bool> HandleRemoteSignOutAsync()
               && Request.Body.CanRead)
             {
-                var form = await Request.ReadFormAsync();
+                var form = await Request.ReadFormAsync(Context.RequestAborted);
```
Why do we need this? This happens automagically without passing in the token.

martincostello replied:
TIL - I didn't realise this was equivalent to the default behaviour with `CancellationToken.None`/`default`.

davidfowl commented on src/Security/Authentication/WsFederation/src/WsFederationHandler.cs:
```diff
@@ -148,7 +148,7 @@ protected override async Task<HandleRequestResult> HandleRemoteAuthenticateAsync
               && Request.Body.CanRead)
             {
-                var form = await Request.ReadFormAsync();
+                var form = await Request.ReadFormAsync(Context.RequestAborted);
```
This too.

davidfowl commented:
In general we shouldn't need to pass the request aborted token reading the request. The server implementation already has knowledge of the so we don't need to pay the cost here.
